src/Runner.php

```diff
@@ -2,87 +2,98 @@
+        return $status_code;
     }
 
     /**
-     * Adds command files to the application
+     * Discovers command classes using CommandFileDiscovery
      *
-     * @param Config $config A Terminus configuration object
-     * @return void
+     * @param string[] $options Elements as follow
+     *        string path      The full path to the directory to search for commands
+     *        string namespace The full namespace associated with given the command directory
+     * @return TerminusCommand[] An array of TerminusCommand instances
      */
-    private function configureApplication(Config $config)
+    private function getCommands(array $options = ['path' => null, 'namespace' => null,])
     {
-        $command_factory = new AnnotatedCommandFactory();
         $discovery = new CommandFileDiscovery();
-        $discovery->setSearchPattern('*Command.php');
-        $subdirectories = array_merge(
-            $this->locateSubdirectories($this->commands_directory, $this->top_namespace),
-            $this->locateSubdirectories($config->get('plugins_dir'))
-        );
-        foreach ($subdirectories as $subdirectory) {
-            $command_files = $discovery->discover($subdirectory['path'], $subdirectory['namespace']);
-            foreach ($command_files as $command_class) {
-                $reflection_class = new \ReflectionClass($command_class);
-                if (!$reflection_class->isAbstract()) {
-                    $command_list = $command_factory->createCommandsFromClass(new $command_class($config));
-                    foreach ($command_list as $command) {
-                        $this->application->add($command);
-                    }
-                }
-            }
-        }
+        $discovery->setSearchPattern('*Command.php')->setSearchLocations(['.']);
+        $commands = $this->fixCommands($discovery->discover($options['path'], $options['namespace']));
+        return $commands;
     }
 
     /**
-     * Recursively discovers all directories underneath a given directory
+     * Removes extraneous /. and \. from the file and class nammes
      *
-     * @param $directory Directory to start search from
-     * @param $namespace Namespace for the given directory
+     * @param string[] $command_info A hash of command file names and paths
+     * @return TerminusCommand[] A fixed and pruned hash of file names and paths
      */
-    private function locateSubdirectories($directory, $namespace = '\\')
+    private function fixCommands($command_info)
     {
-        $subdirectories = [['path' => $directory, 'namespace' => $namespace,]];
-        $dirs = array_filter(glob("$directory/*"), 'is_dir');
-        foreach ($dirs as $dir) {
-            $next_namespace = $namespace.'\\'.str_replace("$directory/", '', $dir);
-            $subdirectories = array_merge($subdirectories, $this->locateSubdirectories($dir, $next_namespace));
-        }
-        return $subdirectories;
+        $commands = array_filter(
```

Comment:
I love it too -- even though I feel bad that you had to write it. ;) You can take it out when you update to annotated commands 1.3.1 and set the search locations to empty instead of ['.'].